plugins/device/rest.go

```diff
@@ -121,6 +123,58 @@ func deviceListHandler(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func deviceEditHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid peer-id passed in url")
+		return
+	}
+
+	req := new(deviceapi.EditDeviceReq)
+	if err := restutils.UnmarshalRequest(r, req); err != nil {
+		logger.WithError(err).Error("Failed to Unmarshal request")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrJSONParsingFailed)
+		return
+	}
+
+	deviceState, exists := deviceapi.DeviceStates[strings.ToLower(req.State)]
+	if !exists {
+		logger.Error("State provided in request doesnot match any state supported by device")
+		errMsg := fmt.Sprintf("invalid state. Supported states are (%v)", getKeys(deviceapi.DeviceStates))
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errMsg)
+		return
+	}
+
+	txn, err := transaction.NewTxnWithLocks(ctx, peerID)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	defer txn.Done()
+	err = deviceutils.SetDeviceState(peerID, req.DeviceName, deviceState)
+	if err != nil {
+		logger.WithError(err).WithField("peerid", peerID).Error("Failed to update device state in store")
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "device state updated")
```

Comment:
The body is not useful. You can pass nil here. It's bad for clients to string matching to check for success case.